Source/WebKit/UIProcess/API/glib/WebKitImage.h.in

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+@API_SINGLE_HEADER_CHECK@
+
+#ifndef WebKitImage_h
+#define WebKitImage_h
+
+#include <gio/gio.h>
+#include <glib-object.h>
+#include <@API_INCLUDE_PREFIX@/WebKitDefines.h>
+
+#if PLATFORM(GTK)
+#include <gtk/gtk.h>
+#endif
+
+G_BEGIN_DECLS
+
+#define WEBKIT_TYPE_IMAGE                       (webkit_image_get_type())
+#if !ENABLE(2022_GLIB_API)
+#define WEBKIT_IMAGE(obj)                       (G_TYPE_CHECK_INSTANCE_CAST((obj), WEBKIT_TYPE_IMAGE, WebKitImage))
+#define WEBKIT_IS_IMAGE(obj)                    (G_TYPE_CHECK_INSTANCE_TYPE((obj), WEBKIT_TYPE_IMAGE))
+#define WEBKIT_IMAGE_CLASS(klass)               (G_TYPE_CHECK_CLASS_CAST((klass),  WEBKIT_TYPE_IMAGE, WebKitImageClass))
+#define WEBKIT_IS_IMAGE_CLASS(klass)            (G_TYPE_CHECK_CLASS_TYPE((klass),  WEBKIT_TYPE_IMAGE))
+#define WEBKIT_IMAGE_GET_CLASS(obj)             (G_TYPE_INSTANCE_GET_CLASS((obj),  WEBKIT_TYPE_IMAGE, WebKitImageClass))
+
+struct _WebKitImageClass {
+    GObjectClass parent_class;
+};
+#endif
+
+WEBKIT_DECLARE_FINAL_TYPE (WebKitImage, webkit_image, WEBKIT, IMAGE, GObject)
+
+WEBKIT_API int          webkit_image_get_width     (WebKitImage *image);
+WEBKIT_API int          webkit_image_get_height    (WebKitImage *image);
+WEBKIT_API guint        webkit_image_get_stride    (WebKitImage *image);
+WEBKIT_API const gchar *webkit_image_get_image_type(WebKitImage *image);
+WEBKIT_API GBytes      *webkit_image_as_rgba_bytes (WebKitImage *image);
+#if PLATFORM(GTK)
+WEBKIT_API GdkTexture  *webkit_image_to_gdk_texture(WebKitImage *image);
```

Comment:
Should we call this `webkit_image_as_gdk_texture()` for consistency with the other method? (cc/ @carlosgcampos)